File: decompiled/3.17.1/be/ehealth/businessconnector/mycarenet/memberdatav2/service/impl/MemberDataServiceImpl.java
package be.ehealth.businessconnector.mycarenet.memberdatav2.service.impl;

import be.ehealth.businessconnector.mycarenet.memberdatav2.service.MemberDataService;
import be.ehealth.businessconnector.mycarenet.memberdatav2.service.ServiceFactory;
import be.ehealth.technicalconnector.config.impl.ConfigurationModuleBootstrap;
import be.ehealth.technicalconnector.exception.TechnicalConnectorException;
import be.ehealth.technicalconnector.exception.TechnicalConnectorExceptionValues;
import be.ehealth.technicalconnector.service.sts.security.SAMLToken;
import be.ehealth.technicalconnector.utils.impl.JaxbContextFactory;
import be.ehealth.technicalconnector.validator.SessionValidator;
import be.ehealth.technicalconnector.ws.domain.GenericRequest;
import be.ehealth.technicalconnector.ws.domain.GenericResponse;
import be.fgov.ehealth.mycarenet.memberdata.protocol.v1.MemberDataConsultationRequest;
import be.fgov.ehealth.mycarenet.memberdata.protocol.v1.MemberDataConsultationResponse;
import javax.xml.soap.SOAPException;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class MemberDataServiceImpl implements MemberDataService, ConfigurationModuleBootstrap.ModuleBootstrapHook {
   private static final Logger LOG = LoggerFactory.getLogger(MemberDataServiceImpl.class);
   private SessionValidator sessionValidator;

   public MemberDataServiceImpl(SessionValidator sessionValidator) {
      this.sessionValidator = sessionValidator;
   }

   public MemberDataServiceImpl() {
      LOG.debug("creating MemberDataServiceImpl for bootstrapping purposes");
   }

   public final MemberDataConsultationResponse consultMemberData(SAMLToken token, MemberDataConsultationRequest request) throws TechnicalConnectorException {
      try {
         this.sessionValidator.validateToken(token);
         GenericRequest service = ServiceFactory.getMemberDataSyncPort(token);
         service.setPayload((Object)request);
         GenericResponse wsResponse = be.ehealth.technicalconnector.ws.ServiceFactory.getGenericWsSender().send(service);
         return (MemberDataConsultationResponse)wsResponse.asObject(MemberDataConsultationResponse.class);
      } catch (SOAPException var5) {
         throw new TechnicalConnectorException(TechnicalConnectorExceptionValues.ERROR_WS, var5, new Object[]{var5.getMessage()});
      }
   }

   public void bootstrap() {
      JaxbContextFactory.initJaxbContext(MemberDataConsultationRequest.class);
      JaxbContextFactory.initJaxbContext(MemberDataConsultationResponse.class);
   }
}

File: src/main/kotlin/org/taktik/connector/business/memberdatav2/service/impl/MemberDataServiceImpl.kt
package org.taktik.connector.business.memberdatav2.service.impl

import be.fgov.ehealth.mycarenet.memberdata.protocol.v1.MemberDataConsultationRequest
import be.fgov.ehealth.mycarenet.memberdata.protocol.v1.MemberDataConsultationResponse
import org.slf4j.LoggerFactory
import org.taktik.connector.business.memberdatav2.service.MemberDataService
import org.taktik.connector.business.memberdatav2.service.ServiceFactory
import org.taktik.connector.technical.config.impl.ConfigurationModuleBootstrap.ModuleBootstrapHook
import org.taktik.connector.technical.exception.TechnicalConnectorException
import org.taktik.connector.technical.exception.TechnicalConnectorExceptionValues
import org.taktik.connector.technical.service.sts.security.SAMLToken
import org.taktik.connector.technical.utils.impl.JaxbContextFactory
import javax.xml.soap.SOAPException

class MemberDataServiceImpl : MemberDataService, ModuleBootstrapHook {
    @Throws(TechnicalConnectorException::class)
    override fun consultMemberData(token: SAMLToken, request: MemberDataConsultationRequest): MemberDataConsultationResponse {
        return try {
            val service =
                ServiceFactory.getMemberDataSyncPort(token)
            service.setPayload(request as Any?)
            val xmlResponse =
                org.taktik.connector.technical.ws.ServiceFactory.getGenericWsSender().send(service)
            val memberDataConsultationResponse = xmlResponse.asObject(MemberDataConsultationResponse::class.java) as MemberDataConsultationResponse

            memberDataConsultationResponse.soapRequest = xmlResponse.request
            memberDataConsultationResponse.soapResponse = xmlResponse.soapMessage

            memberDataConsultationResponse
        } catch (soapException: SOAPException) {
            throw TechnicalConnectorException(TechnicalConnectorExceptionValues.ERROR_WS, soapException, *arrayOf<Any?>(soapException.message))
        }
    }

    override fun bootstrap() {
        JaxbContextFactory.initJaxbContext(MemberDataConsultationRequest::class.java)
        JaxbContextFactory.initJaxbContext(MemberDataConsultationResponse::class.java)
    }
}
